### ISSUE TITLE
Fine-tuning YOLOv8n Model on Two New Classes

### ISSUE THREAD

**Issue by jbdebelle** [ultralytics/ultralytics#13047]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

**Description**
I am working on fine-tuning the yolov8n.pt model to detect two new classes: pick-up truck and armored vehicle. When I pass images of these types of vehicles through the off-the-shelf yolov8n.pt model, they are often detected as one of the following COCO dataset classes:

2: car
5: bus
7: truck

**Goal**
I would like to fine-tune the model to accurately detect pick-up trucks and armored vehicles while leveraging the knowledge from the existing classes (car, bus, truck).

Data Configuration
Here is the configuration in my data.yaml file:

```
path: /path/to/your/dataset  # dataset root directory
train: train/images  # Path to training images directory
val: val/images  # Path to validation images directory

# Classes
names:
  0: pick-up
  1: armored-vehicle
 ```
 
**Request**
I am looking for guidance on how to properly fine-tune the yolov8n.pt model with my custom classes while utilizing the pre-existing knowledge from related classes in the COCO dataset.

**Additional Information**
How should I structure my yaml file for optimal results?
Are there specific training parameters or strategies I should consider to improve detection accuracy for my new classes?
Any recommended practices for leveraging transfer learning effectively in this scenario?
Thank you for your assistance!

**Comment by jbdebelle** [comment]
**Question**
Should I set the classes in my data.yaml file like this to leverage the knowledge from the existing classes?

````
# Classes
names:
  2: pick-up
  7: armored-vehicle
 ````
If so, is it possible to leverage the knowledge from multiple existing classes (car, bus, truck) into a new class?

**Comment by glenn-jocher** [comment]
@jbdebelle hello!

For fine-tuning YOLOv8n on new classes, it's best to define your classes independently in your `data.yaml` file, rather than trying to map them directly to existing COCO class indices. Here’s how you should structure your classes:

```yaml
# Classes
names:
  0: pick-up
  1: armored-vehicle
```

To leverage knowledge from related COCO classes like car, bus, and truck, you can initialize your model with weights from the pre-trained `yolov8n.pt` model. This approach utilizes the learned features for vehicle detection, which can be beneficial for your specific classes.

When training, ensure that your dataset contains examples of pick-up trucks and armored vehicles in various contexts to help the model learn robust features specific to these classes.

Best of luck with your model training! 🚗🛡️
